bayesian/v2/stack_analyses.py

```diff
@@ -0,0 +1,369 @@
+            stack_task_result = stack_result.get("task_result", None)
+            stack_audit = stack_task_result.get("_audit", {})
+
+        if recm_data is not None and "task_result" in recm_data:
+            reco_recommendations = recm_data.get("task_result", {}).get("recommendations", [])[0]
+
+        response_data = {}
+        if stack_task_result is not None:
+            response_data = {
+                "version": stack_audit.get("version", None),
+                "started_at": stack_audit.get("started_at", None),
+                "ended_at": stack_audit.get("ended_at", None),
+                "external_request_id": external_request_id,
+                "registration_status": stack_task_result.get("registration_status", ""),
+                "manifest_file_path": stack_task_result.get("manifest_file_path", ""),
+                "manifest_name": stack_task_result.get("manifest_name", ""),
+                "ecosystem": stack_task_result.get("ecosystem", ""),
+                "unknown_dependencies": stack_task_result.get("unknown_dependencies", ""),
+                "license_analysis": stack_task_result.get("license_analysis", ""),
+                "recommendation": {
+                    "companion": reco_recommendations.get("companion", []),
+                    "manifest_file_path": reco_recommendations.get("manifest_file_path", ""),
+                    "usage_outliers": reco_recommendations.get("usage_outliers", [])
+                },
+                "registration_link": stack_task_result.get("registration_link", ""),
+                "analyzed_dependencies": stack_task_result.get("analyzed_dependencies", [])
+            }
+        else:
+            response_data = {
+                "error": "Enable to fetch the result for request id '{}'".format(
+                    external_request_id)
+            }
+
+        return response_data
+
+    @staticmethod
+    def post():
+        """Handle /api/v2/stack-analyses POST REST API."""
+        # 1. Read mandatory params
+        manifest = request.files.get("manifest") or None
+        file_path = request.form.get("file_path") or None
+        ecosystem = request.form.get("ecosystem") or None
+
+        current_app.logger.info("Mandatory params :: manifest: {} file_path: {} "
+                                "ecosystem: {}".format(manifest, file_path, ecosystem))
+
+        # 2. Read optional params and set default value as per V2 spec.
+        show_transitive = request.form.get("show_transitive") or "true"
+
+        current_app.logger.info("Optional params :: show_transitive: {}".format(show_transitive))
+
+        # 3. Map ecosystem string using static map.
+        ecosystem = StackAnalyses._map_ecosystem(ecosystem)
+
+        current_app.logger.info("Final ecosystem is '{}' after mapping.".format(ecosystem))
+
+        # 4. Validate all post params as per v2 spec.
+        StackAnalyses._validate_post_params(manifest, file_path, ecosystem)
+
```

Comment:
In Python, underscore refers to Private Functions. if so `_validate_post_params` should not be used outside of class. Otherwise remove underscores

Reply:
Its a private function and used only in that class.